server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -126,6 +146,64 @@ public void handle(Request request, Response response) throws Exception {
     response.noContent();
   }
 
+  private void doHandle(DbSession dbSession, SetRequest request) {
+    Optional<ComponentDto> component = searchComponent(dbSession, request);
+    checkPermissions(component);
+
+    PropertyDefinition definition = propertyDefinitions.get(request.getKey());
+
+    if (!request.getFieldValues().isEmpty()) {
+      doHandlePropertySet(dbSession, request, definition, component);
+    } else {
+      validate(request, component);
+      dbClient.propertiesDao().insertProperty(dbSession, toProperty(request, component));
+      if (!component.isPresent()) {
+        settingsChangeNotifier.onGlobalPropertyChange(request.getKey(), persistedValue(request));
```

Comment:
Why notification is not done when the setting is a property set ?
This is ugly, but I think we should return the value in DB (the list of generated ids of fields)